source/retryable-reads/retryable-reads.rst

```diff
@@ -0,0 +1,664 @@
+
+3. Driver release notes MUST make it clear to users that they may need to adjust
+   custom retry logic to prevent an application from inadvertently retrying for
+   too long (see `Backwards Compatibility <#backwards-compatibility>`__ for
+   details).
+
+4. Drivers implementing retryability for their generic command runner for read
+   commands MUST document that ``mapReduce`` will be retried if it is passed as a
+   command to the command runner. These drivers also MUST document the potential
+   for degraded performance given that "Early Failure on Socket Disconnect"
+   feature does not support ``mapReduce``.
+
+Test Plan
+=========
+
+See the `README
+<https://github.com/mongodb/specifications/blob/master/source/retryable-reads/tests/README.rst>`__
+for tests.
+
+At a high level, the test plan will cover executing supported read operations
+within a MongoClient where retryable reads have been enabled, ensuring that
+reads are retried exactly once.
+
+Motivation for Change 
+======================
+
+Drivers currently have an API for the retryability of write operations but not
+for read operations. The driver API needs to be extended to include support for
+retryable behavior for read operations.
+
+Design Rationale 
+=================
+
+The design of this specification is based off the `Retryable Writes
+specification
+<https://github.com/mongodb/specifications/blob/master/source/retryable-writes/retryable-writes.rst#design-rationale>`__.
+It modifies the driver API as little as possible to introduce the concept
+retryable behavior for read operations.
+
+Alternative retry strategies (e.g. exponential back-off, incremental intervals,
+regular intervals, immediate retry, randomization) were considered, but the
+behavior of a single, immediate retry attempt was chosen in the interests of
+simplicity as well as consistency with the design for retryable writes.
+
+See the `future work`_ section for potential upcoming changes
+to retry mechanics.
+
+Backwards Compatibility 
+========================
+
+The API changes to support retryable reads extend the existing API but do not
+introduce any backward breaking changes. Existing programs that do not make use
+of retryable reads will continue to compile and run correctly.
+
+N.B.: Applications with custom retry logic that choose to enable retryable reads
+may need to redo their custom retry logic to ensure that the reads are retried
+as desired. e.g. if an application has custom logic that retries reads n times
+and enables retryable reads, then the application could end up retrying reads up
+to 2n times.
+
+The note above will also apply if an application upgrades to a version of the
+driver where that defaults to enabling retryable reads.
+
+Rejected Designs
+----------------
+
+1. To improve performance on servers without “Early Failure on Socket
+   Disconnect”, we may, we considered using ``killSessions`` to automatically
```

Comment:
I think there's an extra `we may,` here.

Reply:
Thanks, fixed!